services/server/server.go

```diff
@@ -419,6 +427,66 @@ func LoadPlugins(ctx context.Context, config *srvconfig.Config) ([]*plugin.Regis
 	return plugin.Graph(config.DisabledPlugins), nil
 }
 
+type sortInfos []os.FileInfo
+
+func (sf sortInfos) Len() int           { return len(sf) }
+func (sf sortInfos) Less(i, j int) bool { return numFileName(sf[i].Name()) < numFileName(sf[j].Name()) }
+func (sf sortInfos) Swap(i, j int)      { sf[i], sf[j] = sf[j], sf[i] }
+
+func numFileName(name string) int {
+	nums := refileName.FindStringSubmatch(name)
+	if nums == nil || nums[1] == "" {
+		return -1
+	}
+
+	n, _ := strconv.Atoi(nums[1])
+	return n
+}
+
+func vaildConf(info os.FileInfo) bool {
+	if info.Mode().IsRegular() == false {
+		return false
+	}
+	if numFileName(info.Name()) < 0 {
+		return false
+	}
+	return true
+}
+
+// MergeConfd search file in confd , require filename like [0-0]-xx.toml
+// merge conf by Positive sequence into config
+func MergeConfd(config *srvconfig.Config, confd string) error {
```

Comment:
I think config merging logic should reside in `config` package as well.

Reply:
should I mock some .toml file in config directory? I think it's useless when use empty directory .

Reply:
I'm not sure what do you mean by mocking a .toml file. I suggest to split the code up into something like:
```
- config.LoadFromFile(path string) <- loads a Config from a single file.
- config.merge(cfg ...Config) <- merges existing set of Config structs into one final struct 
- config.search(dir, mask string) <- searches config files by mask
- config.LoadFromDir(dir string) <- search+LoadFromFile+merge
```

This way you can test merge/search/load easily.

Reply:
thank your suggestion, I will modify and push it later